### PR TITLE
Add actionLoggerLogMessage hook

### PR DESCRIPTION
**Pull request by PrestaEdit**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add new hook `actionLoggerLogMessage`
| Type?             | new feature
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Related PRs       | autoupgrade: to make.

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
LGTM code wise

The goal is to allow customizing of a log message, right?

Also, can you explain how is this related to autoupgrade, is their an issue that should be linked in the description of this PR ?

**Comment by PrestaEdit** [comment]
Hi,

The goal is to allow a third party module to send the log onto Sentry or another tool needed to make some debug if needed.

The autoupgrade PR will be to add the hook, as required.

**Comment by matthieu-rolland** [comment]
> 

alright 👍

**Review by matthieu-rolland — DISMISSED** [review]
lgtm

File: classes/log/AbstractLogger.php [hunk 1]
```diff
@@ -66,6 +66,15 @@ public function log($message, $level = self::DEBUG)
         if ($level >= $this->level) {
             $this->logMessage($message, $level);
         }
+
+        Hook::exec(
+            'actionLoggerLogMessage',
+            [
+                'message' => $message,
+                'level' => $level,
+                'logged' => $level >= $this->level,
```
Comment (Progi1984) [inline classes/log/AbstractLogger.php]:
```suggestion
                'isLogged' => $level >= $this->level,
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback for naming.

File: install-dev/data/xml/hook.xml [hunk 2]
```diff
@@ -3620,5 +3620,10 @@
       <title>Extra message to display for an empty modules category</title>
       <description>This hook allows to add an extra message to display in the Module manager page when a category doesn't have any module</description>
     </hook>
+    <hook id="actionLoggerLogMessage">
+      <name>actionLoggerLogMessage</name>
+      <title>Allows to make extra action while a log is logged</title>
```
Comment (Progi1984) [inline install-dev/data/xml/hook.xml]:
```suggestion
      <title>Allows to make extra action while a log is triggered</title>
```

:warning: We can trigger a log without logging it (if `$level < $this->level`)

File: install-dev/data/xml/hook.xml [hunk 3]
```diff
@@ -3620,5 +3620,10 @@
...
+      <description>This hook allows to make an extra action while an exception is throw and the logger log it</description>
```
Comment (Progi1984) [inline install-dev/data/xml/hook.xml]:
```suggestion
      <description>This hook allows to make an extra action while an exception is thrown and the logger logs it</description>
```

**Review by Progi1984 — COMMENTED** [review]
Last feedbacks

**Review by Progi1984 — DISMISSED** [review]

**Review by MeKeyCool — DISMISSED** [review]

**Review by matks — CHANGES_REQUESTED** [review]
Dear @PrestaEdit I'm sorry but there is git conflicts for hook.xml file can you please rebase and solve them?

**Comment by matks** [comment]
@PrestaEdit I'm sorry :( you performed a `merge` not a `rebase` so now your branch contains a merge commit and is refused by CI tool Pull Request Validator

**Pull request closed without merging**